app/controllers/manageBusinesses/add/IncomeSourceAddedController.scala

```diff
@@ -22,133 +22,175 @@ import config.{AgentItvcErrorHandler, FrontendAppConfig, ItvcErrorHandler, ShowI
+                      incomeSourceId = incomeSourceId,
+                      showPreviousTaxYears = showPreviousTaxYears,
+                      sessionData = sessionData
+                    )
+                  case _ =>
+                    Future(errorView)
+                }
+            } yield {
+              showView
             }
           }
+
+          result
         case _ =>
-          val agentPrefix = if (isAgent) "[Agent]" else ""
-          Logger("application").error(agentPrefix + s"Unable to retrieve Mongo session data for $incomeSourceType")
-          Future.successful {
-            errorHandler.showInternalServerError()
-          }
+          logger.error(agentPrefix + s"Unable to retrieve Mongo session data for $incomeSourceType")
+          Future(errorView)
       }
-    } recover {
-      case ex: Exception =>
-        Logger("application").error(s"${if (isAgent) "[Agent]" else ""}" +
-          s"Error getting IncomeSourceAdded page: - ${ex.getMessage} - ${ex.getCause}, IncomeSourceType: $incomeSourceType")
-        errorHandler.showInternalServerError()
-    }
-  }
-
-  private def getReportingMethod(maybeAddIncomeSourceData: Option[AddIncomeSourceData]): ChosenReportingMethod = {
-    val (reportingMethodTaxYear1, reportingMethodTaxYear2) = (
-      maybeAddIncomeSourceData.flatMap(_.reportingMethodTaxYear1).orElse(None),
-      maybeAddIncomeSourceData.flatMap(_.reportingMethodTaxYear2).orElse(None)
-    )
-    (reportingMethodTaxYear1, reportingMethodTaxYear2) match {
-      case (Some("A"), Some("A")) | (None, Some("A")) => ChosenReportingMethod.Annual
-      case (Some("Q"), Some("Q")) | (None, Some("Q")) => ChosenReportingMethod.Quarterly
-      case (Some("A"), Some("Q")) | (Some("Q"), Some("A")) => ChosenReportingMethod.Hybrid
-      case (None, None) => ChosenReportingMethod.DefaultAnnual
-      case _ => ChosenReportingMethod.Unknown
     }
   }
 
-  private def handleSuccess(incomeSourceId: IncomeSourceId, incomeSourceType: IncomeSourceType, businessName: Option[String],
-                            showPreviousTaxYears: Boolean, isAgent: Boolean, sessionData: UIJourneySessionData, reportingMethod: ChosenReportingMethod
+  private def handleSuccess(
+                             incomeSourceId: IncomeSourceId,
+                             incomeSourceType: IncomeSourceType,
+                             businessName: Option[String],
+                             showPreviousTaxYears: Boolean,
+                             isAgent: Boolean,
+                             sessionData: UIJourneySessionData
                            )(implicit user: MtdItUser[_], errorHandler: ShowInternalServerError): Future[Result] = {
-    val oldAddIncomeSourceSessionData = sessionData.addIncomeSourceData.getOrElse(AddIncomeSourceData())
-    val updatedAddIncomeSourceSessionData = oldAddIncomeSourceSessionData.copy(journeyIsComplete = Some(true))
+
+    lazy val getNextUpdatesUrl =
```

Comment:
Maybe these should be defs outside the function

Method is longer than 50 lines on my IntelliJ

Reply:
sounds good, moved